lib/rmt/downloader.rb

```diff
@@ -117,7 +120,7 @@ def make_request(remote_file, request_fiber, cache_timestamp = nil)
   end
 
   def copy_from_cache(cache_file, local_file)
-    FileUtils.cp(cache_file, local_file)
+    FileUtils.cp(cache_file, local_file) unless (cache_file == local_file)
```

Comment:
Have you checked if `cache_file == local_file` has any performance issues on larger files? Wouldn't checksum be good enough? There is also [File.identical?](http://ruby-doc.org/core-2.2.0/File.html#method-c-identical-3F) :confused:

Reply:
Both are actually filenames (strings). I should probably rename the variables to `cache_filename` and `local_filename`.